ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Deployment/DeploymentPageDetails.tsx

```diff
@@ -0,0 +1,217 @@
+import React from 'react';
+import {
+    Bullseye,
+    Card,
+    DescriptionList,
+    DescriptionListDescription,
+    DescriptionListGroup,
+    DescriptionListTerm,
+    Divider,
+    Flex,
+    FlexItem,
+    Label,
+    LabelGroup,
+    PageSection,
+    Spinner,
+    Text,
+} from '@patternfly/react-core';
+import { gql, useQuery } from '@apollo/client';
+
+import TableErrorComponent from 'Components/PatternFly/TableErrorComponent';
+import { getDateTime } from 'utils/dateUtils';
+
+export type DeploymentPageDetailsProps = {
+    deploymentId: string;
+};
+
+type DeploymentDetails = {
+    id: string;
+    name: string;
+    cluster: {
+        id: string;
+        name: string;
+    };
+    namespace: string;
+    replicas: number;
+    created: string;
+    serviceAccount: string;
+    type: string;
+    labels: {
+        key: string;
+        value: string;
+    }[];
+    annotations: {
+        key: string;
+        value: string;
+    }[];
+};
+
+const deploymentDetailsQuery = gql`
+    query getDeploymentDetails($id: ID!) {
+        deployment(id: $id) {
+            id
+            name
+            cluster {
+                id
+                name
+            }
+            namespace
+            replicas
+            created
+            serviceAccount
+            type
+            labels {
+                key
+                value
+            }
+            annotations {
+                key
+                value
+            }
+        }
+    }
+`;
+
+// TODO: We want to potentially create reusable Deployment Details components to be shared between Vuln Management, Violations, and Compliance in the future
+// Reference: https://redhat-internal.slack.com/archives/C02MN2N2UG4/p1710184053971889
+function DeploymentPageDetails({ deploymentId }: DeploymentPageDetailsProps) {
+    const { data, previousData, loading, error } = useQuery<{ deployment: DeploymentDetails }>(
+        deploymentDetailsQuery,
+        {
+            variables: {
+                id: deploymentId,
+            },
+        }
+    );
+
+    const deploymentDetailsData = data?.deployment ?? previousData?.deployment;
+
+    return (
+        <>
+            <PageSection component="div" variant="light" className="pf-u-py-md pf-u-px-xl">
+                <Text>View details about this deployment</Text>
+            </PageSection>
+            <Divider component="div" />
+            <PageSection
+                className="pf-u-display-flex pf-u-flex-direction-column pf-u-flex-grow-1"
+                component="div"
+            >
+                {error && (
+                    <TableErrorComponent
+                        error={error}
+                        message="There was an error loading the deployment data"
+                    />
```

Comment:
🤔 Not a request for now, but I wonder if we should rename this component in the future to something like `CenteredError` or similiar, since it isn't always used in the context of a table.